Sources/Views/Products/Search/Scanner/ScanProductSummaryViewAdaptor.swift

```diff
@@ -125,3 +128,14 @@ private func getNovaGroup(from product: RealmOfflineProduct) -> NovaGroupView.No
     }
     return novaGroup
 }
+
+private func getEcoscoreImage(from product: RealmOfflineProduct) -> UIImage? {
+    //guard let co2Impact = product.environmentImpactLevelTags?.first else {
+    guard let ecoscore = product.ecoscore else {
+        return nil
+    }
+    
+    let imageView = EcoscoreImageView(frame: CGRect(origin: .zero, size: CGSize(width: 118, height: 64)))
+    imageView.ecoScore = EcoscoreImageView.Ecoscore(rawValue: ecoscore) ?? .unknown
+    return imageView.image
```

Comment:
This logic is repeated in `getEcoscoreImage(from: Product)`

Reply:
It is doubled as one is for a Priduct and one for a RealmProduct. I was think of changing it to any and do some checking inside.

Reply:
Removed he doubled code